base_partner_sequence/tests/test_base_partner_sequence.py

```diff
@@ -44,9 +44,11 @@ def test_unique_ref_on_write(self):
         ]
         partners = self.env["res.partner"].create(vals)
         self.assertFalse(partners[0].ref == partners[1].ref)
-        partners.write({"ref": False})
+        for partner in partners:
+            partner.write({"ref": False})
```

Comment:
Don't make these changes, my suggestion above (in `def write(self, vals)` of `partner.py`) solves your issue.